### DISCUSSION TITLE
How to rename a file and edit the contents using GIT Bash

### DISCUSSION THREAD

**Discussion by skyhawk7777** [community/community#31509] [New to GitHub] [answered]
Hi, I'm new to gitub/git bash and would really appreciate some help. 

I have successfully used git clone to download a repo to my PC. I now need to change the name of one of the files and also edit some of the the content within the file. 

I used git bash to verify that I can see a copy of the GitHub Repo on my PC - all good.

I'm now trying to copy or rename a file called env.template in one of the subdirectories, to just .env but I cannot find a way of doing this. There are some suggestions to use the mv command but each time I've tried this method it hasn't worked. Could anyone please recommend the best method to do this?

So, in summary, I need to

1. rename, or copy + rename env.template to just .env (dot env)
2. edit the contents of .env to replace it with personalised data

If anyone could assist, I'd be mighty grateful.

Thanks.

**Comment by airtower-luna** [comment, marked as answer]
> rename, or copy + rename env.template to just .env (dot env)

This sounds like the new file shouldn't be part of the repository, is that correct? Usually an `.env` file is for your local environment only. In that case you can just use `cp FROM TO` to copy the file, e.g.: `cp env.template .env`

Then open and edit it whichever editor you prefer.

**Reply by skyhawk7777** [reply]
Hi airtower-luna,

Thank you for your reply.

You are correct the env.template file is no longer required after it has been copied and modified to .env, and the new .env file had it's content personalised.

I did what you suggested, <cp env.template .env> and although no error messages popped up, the new .env file does not show up when I do a dir command in the same directory.

The following screenshot shows the sequence -

![env template](https://user-images.githubusercontent.com/112248116/187738832-512c1ed9-49ed-4cfd-b108-300a2e0d7adb.png)

I'm just wondering where the new file went ?

Thanks,

**Reply by airtower-luna** [reply]
It's hidden, because its name starts with a dot (that's how hidden files work, no magic there). Try `ls -a`. The `-a` means "all", as in "show hidden files, too".

**Reply by skyhawk7777** [reply]
Ah, I see it now, Thank you very much for your kind assistance.

**Reply by airtower-luna** [reply]
Glad I could help! Marking the answer would be appreciated. :slightly_smiling_face:

**Comment by skyhawk7777** [comment]
Thanks for your help

**Reply by airtower-luna** [reply]
I meant marking the answer that solved your problem. :laughing:
